src/backend/distributed/executor/multi_utility.c

```diff
@@ -1661,23 +1687,55 @@ ProcessVacuumStmt(VacuumStmt *vacuumStmt, const char *vacuumCommand)
  * statement needs distributed execution but contains unsupported options.
  */
 static bool
-IsSupportedDistributedVacuumStmt(Oid relationId, VacuumStmt *vacuumStmt)
+IsSupportedDistributedVacuumStmt(VacuumStmt *vacuumStmt, List *vacuumRelationIdList)
 {
 	const char *stmtName = (vacuumStmt->options & VACOPT_VACUUM) ? "VACUUM" : "ANALYZE";
 	bool distributeStmt = false;
+	ListCell *relationIdCell = NULL;
+	int distributedRelationCount = 0;
+	int vacuumedRelationCount = 0;
 
-	if (vacuumStmt->relation == NULL)
+	if (vacuumRelationIdList == NIL)
 	{
 		/* WARN for unqualified VACUUM commands */
 		ereport(WARNING, (errmsg("not propagating %s command to worker nodes", stmtName),
 						  errhint("Provide a specific table in order to %s "
 								  "distributed tables.", stmtName)));
+		return false;
 	}
-	else if (!OidIsValid(relationId) || !IsDistributedTable(relationId))
+
+	vacuumedRelationCount = list_length(vacuumRelationIdList);
+
+	foreach(relationIdCell, vacuumRelationIdList)
 	{
-		/* Nothing to do here; relation no longer exists or is not distributed */
+		Oid relationId = lfirst_oid(relationIdCell);
+		if (OidIsValid(relationId) && IsDistributedTable(relationId))
+		{
+			distributedRelationCount++;
+		}
 	}
-	else if (!EnableDDLPropagation)
+
+	/*
+	 * There can only be one distributed relation in the vacuum statement.
+	 * If there is no distributed relation, then we silently fall back
+	 * since there is nothing to do for us. Error cases are there is more
+	 * than one distributed relation, or there is one distributed relation
+	 * along with other relations.
+	 */
+	if (distributedRelationCount == 0)
+	{
+		/* Nothing to do here; there is no distributed relation */
+		return false;
+	}
+	else if (distributedRelationCount > 0 && vacuumedRelationCount > 1)
+	{
+		ereport(ERROR, (errmsg("vacuuming multiple tables at the same time "
```

Comment:
I think I remembered why everything should be `WARNING`… We've already run the `VACUUM` against the local table at this point, right? But `VACUUM` is _never_ transactional, so throwing an error doesn't actually roll back anything, so I used a `WARNING`. For consistency, can we keep that behavior?

Reply:
Also `VACUUM`ing multiple tables at the same time _is_ supported, just not by Citus. Shouldn't this be phrased like "`VACUUM`ing a distributed table at the same time as other tables is not supported" or something?